horovod/_keras/__init__.py

```diff
@@ -167,32 +185,83 @@ def allreduce_grads():
                             # `IndexedSlices`.
                             return [tf.identity(g) for g in averaged_gradients]
 
+            def trim_last_curly_brace(s):
+                if s[-1] != "}":
+                    raise AssertionError(f"Expected last character in \"{s}\" to be \"}}\", but got \"{s[-1]}\"")
+                return s[:-1]
+
+            def profile_start():
+                if self._profile_frequency == 0:
+                    return tf.no_op()
+                return tf.cond(tf.equal(self.profile_counter, self._profile_frequency - 1), log_comm_start, tf.no_op)
+
+            def log_comm_start():
+                profile_base_info = trim_last_curly_brace(get_profile_info("communication", "B"))
+                self.start_timestamp = tf.timestamp() * 1e6
+                return tf.print(profile_base_info, ", \"ts\": ", self.start_timestamp, "}", sep="", output_stream="file:///profile/pedl_profiling.log")
```

Comment:
Should we make the `output_stream` file path configureable?

Reply:
I have a few concerns with configuring this. Not configuring this allows us to easily build features around this. Would we want users to be able to configure different files for our Horovod stuff and PEDL stuff? We don't necessarily want to make everything configureable, and think it's fine to not configure this for now. If we want to make it configureable in the future it should not be bad.